source/MaterialXCore/Interface.cpp

```diff
@@ -154,6 +154,21 @@ Edge Input::getUpstreamEdge(ConstMaterialPtr material, size_t index) const
     return NULL_EDGE;
 }
 
+GeomPropPtr Input::getGeomProp() const
```

Comment:
Does this need to be in a more restrictive place as not all interface elements can have geometric properties?

Reply:
It's not an _Interface_ method. It's an _Input_ method, The Input class is defined in this file.

Reply:
Yes. I was looking at the file name instead of the class code. My mistake :)